### ISSUE TITLE
Misconceptions between the sdk and the api as regards to email validation

### ISSUE THREAD

**Issue by nmargaritis** [mailgun/mailgun-php#631]
Hi, we have used your library in order to integrate mailgun email validation in a Laravel project.

Following the documentation [API documentation](https://documentation.mailgun.com/en/latest/api-email-validation.html#email-validation) from here and the php example, we could not get the validation to work.

Now upon this I would like to state a few points.
- Firstly your sdk is bound to version 3 of the mailgun api, which is not mentioned anywhere. Please update your README file. The examples of the API are actually for v4
- The api documentation advertises this library although the lib does not offer support for v4.... The lack of support is not even in the documentation, aside from this library.... [Libraries](https://documentation.mailgun.com/en/latest/libraries.html#php)
- Due to the mismatch of your sdk and the api versions that are not discussed, you can not perform an email address validation! I will explain why below.
- Fix your sdk errors to contain the actual response from the API
- If there is a way to use v4, add it on your README file.

```
$mgClient = new Mailgun('PRIVATE_API_KEY');
$validateAddress = 'foo@mailgun.net';
$result = $mgClient->get("address/validate", array('address' => $validateAddress));
```
Due to the mismatch of your sdk and the api versions that are not discussed, you can not perform an email address validation
--

As you can see from this example, the Mailgun client is supposed to be initiated with the private key, however the `/v3/address/validation` is awaiting for a public key and not a private key, otherwise the request fails with 401.
```
$mgClient = new Mailgun('PRIVATE_API_KEY');
```
While attempting this call with your library, one gets the following message

> Your credentials are incorrect

So they then try to figure out why the private key is wrong.... The actual message from the api is that a public key has to be used, which your library hides and returns the `Your credentials are incorrect` message.

There is a high chance that all these reports about `Your credentials are incorrect` that can be found as library issues, have to do with similar misconceptions from both the part of the API and SDK.

Bottom line is that, we decided to go with Guzzle.
